analysis/loadprogram/state.go

```diff
@@ -173,3 +175,50 @@ func (wps *State) IncrementAndTestAlarms() bool {
 func (wps *State) TestAlarmCount() bool {
 	return wps.Config.MaxAlarms <= 0 || wps.numAlarms.Load() < int32(wps.Config.MaxAlarms)
 }
+
+func allAstFiles(dirs []string, fset *token.FileSet, pkgs []*packages.Package) ([]*ast.File, error) {
+	var files []*ast.File
+
+	for _, dir := range dirs {
+		if err := filepath.Walk(dir, func(path string, info fs.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+
+			if info.IsDir() {
+				parsedDir, err := parser.ParseDir(fset, path, nil, parser.ParseComments)
```

Comment:
We can use this as a hack for now, but I think we should look further into getting all the files we need, I hope we don't really have to parse file twice.